tests/particle_test.cc

```diff
@@ -1383,10 +1380,10 @@ TEST_CASE("Particle is checked for 2D case", "[particle][2D]") {
     // check if the correct amount of material ids were added to node and if
     // their indexes are correct
     std::vector<unsigned> material_ids = {0, 1};
-    for (int j = 0; j < nodes.size(); ++j) {
-      REQUIRE(nodes[j]->material_ids().size() == 2);
+    for (auto node_itr : nodes) {
```

Comment:
This is not an iterator it's the node so please use for (const auto& node : nodes) {